File: java_ws/ProgettoJava/src/attivita/Servizio.java
package attivita;

import java.time.LocalTime;

public class Servizio extends Attivita{
	final TipoServizio tipo;	
	boolean suPrenotazione;
	
	//Costruttori
	public Servizio(String nome, LocalTime orarioApertura, LocalTime orarioChiusura,
			int valutazione, int affollamento, TipoServizio tipo, boolean suPrenotazione)
			throws ValutazioneOutOfRangeException, AffolamentoOutOfRangeException {
		super(nome, orarioApertura, orarioChiusura, valutazione, affollamento);
		this.tipo = tipo;
		this.suPrenotazione = suPrenotazione;
	}
	public Servizio(String nome, LocalTime orarioApertura, LocalTime orarioChiusura,
			TipoServizio tipo, boolean suPrenotazione)
			throws ValutazioneOutOfRangeException, AffolamentoOutOfRangeException {
		this(nome, orarioApertura, orarioChiusura, 0, 0,  tipo, suPrenotazione);
	}
	
	//Metodi
	@Override
	public String toString() {
		return "Servizio " + tipo + ": " + super.toString() +
			   " 	-> Richiesta prenotazione: " + (suPrenotazione? "SI\n" : "NO\n");
	}
	
	public boolean isSuPrenotazione() {
		return suPrenotazione;
	}
	
	public TipoServizio getTipo() {
		return tipo;
	}
}

File: cpp_ws/ProgettoCpp/src/Servizio.cpp
#include "Servizio.h"

namespace attivita {

Servizio::Servizio(string m, Time oA, Time oC, int v, int a, TipoServizio tS, bool sP)
		:Attivita::Attivita(m, oA, oC, v, a), tipo(tS), suPrenotazione(sP){}

Servizio::Servizio(string m, Time oA, Time oC, TipoServizio tN, bool sP)
		:Servizio::Servizio(m, oA, oC, 0, 0, tN, sP){}

string Servizio::toString(){
	return "Servizio di "  + enumToString[static_cast<int>(tipo)] + ": " + Attivita::toString() +
			" 	-> Richiesta prenotazione: " + (suPrenotazione? "SI\n" : "NO\n");
}

Servizio::~Servizio() {}

} /* namespace attivita */
